Examples/DpdkExample-FilterTraffic/main.cpp

```diff
@@ -162,7 +162,7 @@ void listDpdkPorts()
 void prepareCoreConfiguration(std::vector<pcpp::DpdkDevice*>& dpdkDevicesToUse,
                               std::vector<pcpp::SystemCore>& coresToUse, bool writePacketsToDisk,
                               const std::string& packetFilePath, pcpp::DpdkDevice* sendPacketsTo,
-                              AppWorkerConfig workerConfigArr[], int workerConfigArrLen, uint16_t rxQueues)
+                              std::vector<AppWorkerConfig>& workerConfigArr, int workerConfigArrLen, uint16_t rxQueues)
```

Comment:
Doesn't this still work as an unknown bound array? It should be fine to be passed the `data()` ptr from the vector on the call site?

If it is kept as a vector ref, there is no point to the `workerConfigArrLen`.

Reply:
Yes, we could pass the `data()` pointer, but there's no strong reason to do so. The `prepareCoreConfiguration` function iterates over `coresToUse` and fills the `workerConfigArr`, whose size is determined by `coresToUse.size()`. Thus, the logic of the code remains unchanged—we still iterate over its size and populate it.

As for `workerConfigArrLen`, it's an interesting case because it was unused even before this patch. Perhaps we should address this in a separate PR that removes unused parameters?